registry/deploying.md

```diff
@@ -1,94 +1,246 @@
+
+    ```bash
+    $ docker stop registry
+    ```
+
+3.  Restart the registry, directing it to use the TLS certificate. This command
+    bind-mounts the `certs/` directory into the container at `/certs/`, and sets
+    environment variables that tell the container where to find the `domain.crt`
+    and `domain.key` file.
+
+    ```bash
+    $ docker run -d \
+      -p 5000:5000 \
+      --restart=always \
+      --name registry \
       -v `pwd`/certs:/certs \
       -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
       -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
       registry:2
+    ```
 
-You should now be able to access your registry from another docker host:
+4.  Docker clients can now pull from and push to your registry using its
+    external address. The following commands demonstrate this:
 
-    docker pull ubuntu
-    docker tag ubuntu myregistrydomain.com:5000/ubuntu
-    docker push myregistrydomain.com:5000/ubuntu
-    docker pull myregistrydomain.com:5000/ubuntu
+    ```bash
+    $ docker pull ubuntu:16.04
+    $ docker tag ubuntu:16.04 myregistrydomain.com:5000/my-ubuntu
+    $ docker push myregistrydomain.com:5000/my-ubuntu
+    $ docker pull myregistrydomain.com:5000/my-ubuntu
+    ```
 
-#### Gotcha
+#### Use an intermediate certificate
 
-A certificate issuer may supply you with an *intermediate* certificate. In this case, you must combine your certificate with the intermediate's to form a *certificate bundle*. You can do this using the `cat` command:
+A certificate issuer may supply you with an *intermediate* certificate. In this
+case, you must concatenate your certificate with the intermediate certificate to
+form a *certificate bundle*. You can do this using the `cat` command:
 
-    cat domain.crt intermediate-certificates.pem > certs/domain.crt
+```bash
+cat domain.crt intermediate-certificates.pem > certs/domain.crt
+```
+
+You can use the certificate bundle just as you use the `domain.crt` file in
+the previous example.
 
-### Let's Encrypt
+### Support for Let's Encrypt
 
-The registry supports using Let's Encrypt to automatically obtain a browser-trusted certificate. For more
-information on Let's Encrypt, see [https://letsencrypt.org/how-it-works/](https://letsencrypt.org/how-it-works/) and the relevant section of the [registry configuration](configuration.md#letsencrypt).
+The registry supports using Let's Encrypt to automatically obtain a
+browser-trusted certificate. For more information on Let's Encrypt, see
+[https://letsencrypt.org/how-it-works/](https://letsencrypt.org/how-it-works/)
+and the relevant section of the
+[registry configuration](configuration.md#letsencrypt).
 
-### Alternatives
+### Use an insecure registry (testing only)
 
-While rarely advisable, you may want to use self-signed certificates instead, or use your registry in an insecure fashion. You will find instructions [here](insecure.md).
+It is possible to use a self-signed certificate, or to use our registry
```

Comment:
You can still setup verification with a self-signed certificate that isn't used for testing.